agc/asm.c

```diff
@@ -19,331 +19,358 @@
+  Contact:   Inokentiy Babushkin
   Reference: http://www.ibiblio.org/apollo
   Mods:      08/31/08 OH. Began.
              07/12/16 IB. Began refactoring to include in radare2-extras.
+             07/17/16 IB. Annotated disassembler source and added symbolic
+                          constants.
 */
 
 #include <stdbool.h>
-#include <stdio.h>
-#include "asm_agc.h"
+#include "asm.h"
 
-/* 
- * We need to find out what the logic behind the modifications done by
- * ShowAddressContents() is. Since that function performs some lookup
- * on the internal state of the AGC *simulator*, it is very well possible
- * we can simplify things a fair bit. This would also mean that we might be
- * able to reduce the decoding work to interpreting the EXTEND instruction
- * as a prefix, since that's it's only function as far as we know so far.
- *
- * However, if the assembling/disassembling doesn't yield the desired
- * results, we will be in deep trouble already.
- *
- * So far, the best thing to do is to document the functionality of the
- * different static variables employes:
- * * sCurrentZ: the current address to be disassembled.
- *   It appears that they are handled differently depending on their value.
- *   For instance, the bank gets computed using them (duh).
- * * sBank: the memory bank the address maps to.
- *   Memory bank addressing was used because of storage shortage for
- *   adresses, which led to this workaround. Banks have different sizes
- *   (1K or 256 bytes), thus they get used differently in the code.
- * * sValue: the memory value at the location denoted by the address.
- * * sErasable: whether the data we access is in erasable memory.
- * * sFixed: whether we are reading from ROM.
- *
- * But why are we using two variables to denote mutually exclusive flags?
- *
- * We can get the current address from the RAsm structure pointer passed
- * to disassemble(). Thus, we can compute erasable and fixed.
- *
- * NOP's seem to be encoded in a position-depend way, but that's all we
- * need to handle specially.
- */
+const char *agc_mnemonics[59] = {
+    // 3-bit opcode, no operand
+    "com", "ddoubl", "double", "dtcb", "dtcf", "extend", "inhint", "noop",
+    "ovsk", "relint", "resume", "return", "tcaa", "xlq", "xxalq", "zl",
+    // 3-bit opcode, w/ operand
+    "tc\t%04o", "ccs\t%04o", "tcf\t%04o", "das\t%04o", "lxch\t%04o",
+    "incr\t%04o", "ads\t%04o", "ca\t%04o", "cs\t%04o", "index\t%04o",
+    "dxch\t%04o", "ts\t%04o", "xch\t%04o", "ad\t%04o", "mask\t%04o",
+    // 4-bit opcode, no operand
+    "dcom", "resume", "square", "zq",
+    // 4-bit opcode, w/ small operand
+    "read\t%03o", "write\t%03o", "rand\t%03o", "wand\t%03o", "ror\t%03o",
+    "wor\t%03o", "rxor\t%03o", "edrupt\t%03o",
+    // 4-bit opcode, w/ normal operand
+    "dv\t%04o", "bzf\t%04o", "msu\t%04o", "qxch\t%04o", "aug\t%04o",
+    "dim\t%04o", "dca\t%04o", "dcs\t%04o", "index\t%04o", "su\t%04o",
```

Comment:
Dont use tabs or that will break the visual mode. Only add one space. You can use asm.tabs to specify a diffeeent indent between mnemonic and operands